flycheck.el

```diff
@@ -3083,8 +3089,12 @@ non-whitespace character of the error line, if ERR has no error column."
 (defun flycheck-error-format-message-and-id (err)
   "Format the message and id of ERR as human-readable string."
   (let ((id (flycheck-error-id err))
-        (message (flycheck-error-message err)))
-    (if id (format "%s [%s]" message id) message)))
+        (filename (flycheck-error-filename err)))
+    (concat (when (and filename (not (equal filename (buffer-file-name))))
+              (format "In \"%s\": " (file-relative-name filename default-directory)))
```

Comment:
Maybe add a newline after the colon?

Reply:
Should we show the path relative to the default directory, or to the current file? (I'm not sure whether default-directory here is the checker's `:working-directory` or the current buffer's folder)

Reply:
I'll add the newline and check whether the default directory is the right one here.

Reply:
I checked and the default-directory is the correct one, i.e. it's the directory of the current buffer's file.